plugin/storage/cassandra/spanstore/reader.go

```diff
@@ -110,14 +112,19 @@ type SpanReader struct {
 	operationNamesReader operationNamesReader
 	metrics              spanReaderMetrics
 	logger               *zap.Logger
+	tracer               trace.Tracer
 }
 
 // NewSpanReader returns a new SpanReader.
 func NewSpanReader(
 	session cassandra.Session,
 	metricsFactory metrics.Factory,
 	logger *zap.Logger,
+	tracer trace.Tracer,
 ) *SpanReader {
+	if tracer == nil {
+		tracer = otel.GetTracerProvider().Tracer("cSpanStore.SpanReader")
+	}
```

Comment:
if you're passing a tracer as argument it's expected to be non-null. Don't need this check / fallback.